aip/general/0130.md

```diff
@@ -0,0 +1,88 @@
+| **Standard Methods**               |                                                                                                                                                                                                               |                                                    |                                 |                 |                |                 |
+| Standard collection methods        | Operate on a collection of resources (List or Create).                                                                                                                                                        | [AIP-121][], [AIP-132][], [AIP-133][]              | automatable                     | automatable     | automatable    | automatable     |
+| Standard resource methods          | Fetch or mutate a single resource (Get, Update, Delete).                                                                                                                                                      | [AIP-121][], [AIP-131][], [AIP-134][], [AIP-135][] | automatable                     | automatable     | automatable    | automatable     |
+| Batch resource methods             | Fetch or mutate multiple resources in a collection by name.                                                                                                                                                   | [AIP-231][], [AIP-233][], [AIP-234][], [AIP-235][] | may be used to optimize queries | automatable     | automatable    | automatable     |
+| Aggregated list methods            | Fetch or mutate multiple resources of the same type across multiple collections.                                                                                                                              | [AIP-159][]                                        | not useful nor automable        | automatable     | automatable    | automatable     |
+| **Custom Fetch Methods**           |                                                                                                                                                                                                               |                                                    |                                 |                 |                |                 |
+| Custom collection fetch methods    | Fetch information across a collection that cannot be expressed via a standard method.                                                                                                                         | [AIP-136][]                                        | automatable                     | automatable     | automatable    | automatable     |
```

Comment:
Are custom collection and resource fetch methods automatable in Terraform? Is there an example of this now?

Reply:
I was imagining that any custom fetch method could become a Terraform data source (the request parameters would be the resource itself).

But I can check with the Terraform team. But I guess to be on the safe side I'll put this in "handwritten".